src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -168,6 +169,19 @@
         "dollar_quote", CodeSegment, name="dollar_quoted_literal", type="literal"
     ),
     SimpleGeometryGrammar=AnyNumberOf(Ref("NumericLiteralSegment")),
+    # N.B. this MultilineConcatenateDelimiterGrammar is only created
+    # to parse multiline-concatenated string literals
+    # and shouldn't be used in other contexts.
+    # In general let the parser handle newlines and whitespace.
+    MultilineConcatenateNewline=NamedParser(
+        "newline",
+        NewlineSegment,
+        name="newline",
+        type="newline",
+    ),
+    MultilineConcatenateDelimiterGrammar=AnyNumberOf(
+        Ref("MultilineConcatenateNewline"), min_times=1, allow_gaps=False
+    ),
```

Comment:
So the reason
```sql
SELECT 'foo'

'bar';
```
didn't parse before is that I needed to allow mulitple newlines as the delimiters.

I also had to define `MultilineConcatenateNewline` as the parser didn't like me using a `NamedParser` within the `AnyNumberOf`.

The `allow_gaps` is to force the delimiter to be greedy with multiple newlines (n.b. still allows whitespace, see unit tests) so all good.